### ISSUE TITLE
Bug / Feature -  Convert ifc element class to ifc type,  use ifc modifiers to make types.

### ISSUE THREAD

**Issue by macedonianluke** [IfcOpenShell/IfcOpenShell#3406]
Can we please make a method for turning an ifc door element class into an ifc door type? 
And or is there also a way of using the door ifc modifier to create a door type?  
Should there be a simple method of say right click convert class to type or similar?
Using door creation as an example should work for all ifc elements that you need to create new types for.

**Comment by theoryshaw** [comment]
> Can we please make a method for turning an ifc door element class into an ifc door type?



https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/0405412b-7f08-4559-b819-9f4af323e061


> And or is there also a way of using the door ifc modifier to create a door type?


https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/891409db-cbf3-492a-a1e9-ec58931c16d4




> Should there be a simple method of say right click convert class to type or similar?

https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/0405412b-7f08-4559-b819-9f4af323e061

> Using door creation as an example should work for all ifc elements that you need to create new types for.

Not sure i follow this request.

**Comment by macedonianluke** [comment]
Okey so you can with your process above.  Sorry didnt no that.  Maybe my request is more a simplification of the process.   Eg right click convert element to type. cheers thanks Ryan.

**Comment by Gorgious56** [comment]
This may be a XY problem that we should try to fix the root cause of, could you elaborate on the workflow that makes you need this feature ?

**Comment by Moult** [comment]
I think this is a valid workflow - say working on a bespoke design, you might then realise a part needs to be modularised (i.e. turned into a type). But it doesn't happen very often.

@macedonianluke another way of doing it is to Launch the type manager with your occurrence selected, then use the "Custom Mesh" type template and press save new type. This will create a new type with the geometry of the occurrence. I think that is easy enough?

![2023-07-10-172115_1090x534_scrot](https://github.com/IfcOpenShell/IfcOpenShell/assets/88302/3ec4147f-88fb-449e-8ba0-4c075661ac2d)

**Comment by Gorgious56** [comment]
Yes I actually use this workflow a lot (too much ?), too and the process is indeed a bit cumbersome. I've been thinking of ways to improve it because right now it involves too many clicks and muscle memory :) I think the "Custom Mesh" option in the BIM tool is straightforward enough.

BTW I think it hasn't been mentioned but you can create an IfcDoor modifier directly within this interface

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/25156105/e3af98c5-8abc-4fdc-90e5-c8dbe302a66a)
